src/Symfony/Component/VarDumper/Caster/OpenSslCaster.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Caster;
+
+use Symfony\Component\VarDumper\Cloner\Stub;
+
+/**
+ * @author Alexandre Daubois <alex.daubois@gmail.com>
+ */
+class OpenSslCaster
+{
+    public static function castOpensslX509($h, array $a, Stub $stub, bool $isNested): array
+    {
+        $stub->cut = -1;
+        $info = openssl_x509_parse($h, false);
+
+        $pin = openssl_pkey_get_public($h);
+        $pin = openssl_pkey_get_details($pin)['key'];
+        $pin = \array_slice(explode("\n", $pin), 1, -2);
+        $pin = base64_decode(implode('', $pin));
+        $pin = base64_encode(hash('sha256', $pin, true));
+
+        $a += [
+            'subject' => new EnumStub(array_intersect_key($info['subject'], ['organizationName' => true, 'commonName' => true])),
+            'issuer' => new EnumStub(array_intersect_key($info['issuer'], ['organizationName' => true, 'commonName' => true])),
+            'expiry' => new ConstStub(date(\DateTimeInterface::ISO8601, $info['validTo_time_t']), $info['validTo_time_t']),
+            'fingerprint' => new EnumStub([
+                'md5' => new ConstStub(wordwrap(strtoupper(openssl_x509_fingerprint($h, 'md5')), 2, ':', true)),
+                'sha1' => new ConstStub(wordwrap(strtoupper(openssl_x509_fingerprint($h, 'sha1')), 2, ':', true)),
+                'sha256' => new ConstStub(wordwrap(strtoupper(openssl_x509_fingerprint($h, 'sha256')), 2, ':', true)),
+                'pin-sha256' => new ConstStub($pin),
+            ]),
+        ];
+
+        return $a;
+    }
+
+    public static function castOpensslAsymmetricKey($h, array $a, Stub $stub, bool $isNested): array
+    {
+        $info = openssl_pkey_get_details($h);
+
+        $a += [
+            Caster::PREFIX_VIRTUAL.'type' => new ConstStub($info['type'], $info['type']),
+            Caster::PREFIX_VIRTUAL.'bits' => new ConstStub($info['bits'], $info['bits']),
```

Comment:
ConstStub are for ... consts, there are non here

Reply:
I think @alexandre-daubois thought that everything needs to be wrapped in a stub...

Reply:
Indeed, I was a little too “inspired” by  [this](https://github.com/symfony/symfony/blob/4ed9d032c7b61ae49d526565bbc8fdbcf76d74c3/src/Symfony/Component/VarDumper/Caster/ResourceCaster.php#L66). 🤦 Removing all the useless stubs.